audio/softsynth/mt32.cpp

```diff
@@ -169,12 +169,13 @@ int MidiDriver_MT32::open() {
 	debug(4, _s("Initializing MT-32 Emulator"));
 
 	Common::File controlFile;
-	if (!controlFile.open("CM32L_CONTROL.ROM") && !controlFile.open("MT32_CONTROL.ROM"))
-		error("Error opening MT32_CONTROL.ROM / CM32L_CONTROL.ROM. Check that your Extra Path in Paths settings is set to the correct directory");
-
 	Common::File pcmFile;
-	if (!pcmFile.open("CM32L_PCM.ROM") && !pcmFile.open("MT32_PCM.ROM"))
-		error("Error opening MT32_PCM.ROM / CM32L_PCM.ROM. Check that your Extra Path in Paths settings is set to the correct directory");
+	if (!controlFile.open("CM32L_CONTROL.ROM") || !pcmFile.open("CM32L_PCM.ROM")) {
+		debug("Error opening CM32L_CONTROL.ROM / CM32L_PCM.ROM. Falling back to MT32");
+
+		if (!controlFile.open("MT32_CONTROL.ROM") || !pcmFile.open("MT32_PCM.ROM"))
```

Comment:
```suggestion
		if (!controlFile.open("MT32_CONTROL.ROM") && !pcmFile.open("MT32_PCM.ROM"))
```

Reply:
Like above using `||` is expected to make it fail as soon as there is an error.
Anyway, using open here when one CM32L file exists will trigger an assertion fault in open because one of the file is opened.
Both files should be closed before trying again.

Reply:
Same as above, if any of the two files can't be opened, the check should fail. The following is the equivalent way of writing the same (edited) check, using boolean math:

```suggestion
		if (!(controlFile.open("MT32_CONTROL.ROM") || pcmFile.open("MT32_PCM.ROM")))
```

Reply:
You're correct. The original code is OK here